sp/Pedestrian.py

```diff
@@ -306,52 +306,20 @@ def __init__(self, pid, name, start_state, frenet_state, yaw, path, debug_shdata
         self.speed_qualifier = speed_qualifier
         self.reference_speed = reference_speed
         self.scenario_vehicles = scenario_vehicles
+        self.use_collision_point = use_collision_point
         
         self.current_waypoint = 0.0
 
         self.current_path_node = 0
     
     def tick(self, tick_count, delta_time, sim_time):  
         ped_path = [(n.x, n.y) for n in self.path]
-
-        Pedestrian.tick(self, tick_count, delta_time, sim_time)
-
-        # Check if collision_vid is provided
-        if self.collision_vid is not None:
-            # Try to get the collision vehicle, if not found, raise a KeyError
-            collision_vehicle = None
-            try:
-                collision_vehicle = self.scenario_vehicles[self.collision_vid]
-                vehicle_pos = np.array([collision_vehicle.state.x, collision_vehicle.state.y])
-                vehicle_vel = np.array([collision_vehicle.state.x_vel, collision_vehicle.state.y_vel])
-            except KeyError:
-                raise KeyError(f"collision vehicle with vid {self.collision_vid} not found in scenario")
-
-            # Calculate collision details if vehicle exists
-            if collision_vehicle:
-                if self.get_collision_pt(vehicle_pos, vehicle_vel, self.path) is not None:
-                    collision_pt, collision_segment_prev_node, collision_segment_next_node = self.get_collision_pt(vehicle_pos, vehicle_vel, self.path)
-
-                    # Euclidean distance between vehicle and collision point
-                    collision_vehicle_dist_to_collision = np.sqrt(np.sum((collision_pt - vehicle_pos) ** 2))
-                    if collision_vehicle.state.s_vel == 0:
-                        time_to_collision = float('inf')
-                    else:
-                        time_to_collision = collision_vehicle_dist_to_collision / collision_vehicle.state.s_vel
-                else:
-                    collision_pt = None
-                    collision_segment_next_node = None
-                    collision_segment_prev_node = None
-                    time_to_collision = None
-        else:
-            # If collision_vid is None, set default None values to make pedestrian behave with default speed parameters from scenario file
-            time_to_collision = None
-            collision_pt = None
-            collision_segment_prev_node = None
-            collision_segment_next_node = None
+        time_to_collision = None
+        
+        Pedestrian.tick(self, tick_count, delta_time, sim_time)  
 
         # Proceed with follow_path, using None values if no collision or vehicle exists
-        self.follow_path(delta_time, sim_time, self.path, time_to_collision, collision_pt, collision_segment_prev_node, collision_segment_next_node, self.speed_qualifier, self.reference_speed)
+        self.follow_path(delta_time, sim_time, self.path, time_to_collision, self.speed_qualifier, self.reference_speed)
```

Comment:
does the `self.reference_speed` ever change? Can it just be kept as a field in Actor and passed to the super constructor on line 300? Same for the `speed_qualifier`